darwin/future/core/types/common.py

```diff
@@ -1,13 +1,20 @@
 from __future__ import annotations
 
-from typing import Any, Dict, List, Union
+from typing import Any, Dict, List, Mapping, Protocol, Union
 
 from darwin.future.data_objects import validators as darwin_validators
-from darwin.future.data_objects.typing import UnknownType
 
 JSONType = Union[Dict[str, Any], List[Dict[str, Any]]]  # type: ignore
 
 
+class Implements_str(Protocol):
+    def __str__(self) -> str:
+        ...
+
+
+Stringable = Union[str, Implements_str]
```

Comment:
I'm not entirely sure why this union has to exist, but I couldn't get the type system to properly work without it. One would expect implements_str to encompass str natively, but it doesn't.